gamedev/part1.md

```diff
@@ -0,0 +1,428 @@
+# Game Dev
+## Part 1: Drawing on the Screen
+<a href="README.md#instructions">Back</a>
+
+### How do game visuals work?
+
+In games and movies, it looks like the things on the screen on the moving. The movement you see is actually just a bunch of images with small changes played back in rapid succession, and then your brain interprets this as movement.
+
+A classic example of this is "The Horse In Motion", which was made in 1878 by taking photos of a racehorse at different points in time. Played back, the horse appears to move.
+
+<img src="https://upload.wikimedia.org/wikipedia/commons/7/73/The_Horse_in_Motion.jpg" />
+
+*Frames of The Horse In Motion, from <a href="https://commons.wikimedia.org/wiki/File:The_Horse_in_Motion.jpg" target="_blank">Wikimedia</a>*
+
+<img src="https://upload.wikimedia.org/wikipedia/commons/d/dd/Muybridge_race_horse_animated.gif" />
+
+*The frames, played back, from <a href="https://en.wikipedia.org/wiki/File:Muybridge_race_horse_animated.gif" target="_blank">Wikimedia</a>*
+
+In a game, the same thing applies. The movement in a game will be made by the computer drawing slightly different images together fast enough to give the illusion of motion. Unlike in real life, where objects move on its own for us to capture, in a game, the computer needs to figure out where to put everything in the next frame before it can draw it.
+
+In general, a minimum **frame rate** of 18 frames per second is required for images to appear to be moving to the human brain. Games tend to run at higher rates of 30 or 60 frames per second because reaction time can be central to gameplay, making it important to have more detailed motion. This means that 60 times per second, the computer will run a program to recalculate the new positions of objects, draw an image, and display it on the screen.
+
+### Javascript crash course
+
+So we know we're going to have to figure out where to move each object in every update in every frame. Let's see what tools we have at our disposal!
+
+Our game will be made in Javascript using a graphics library called p5.js. Here are the main parts of Javascript you will be using to make your game:
```

Comment:
Here might be a good place to go into p5.js, Javascript and how its different from Java.